kcidb/test_oo.py

```diff
@@ -203,3 +204,225 @@ def test_two_object(self):
             '>checkout#>build#>test#',
             {"test", "build", "checkout", "revision"}, 2
         )
+
+
+class KCIDBTraversingTestCase(kcidb.unittest.TestCase):
+    """Test case for traversing objects."""
+
+    # Conform to unittest conventions, pylint: disable=invalid-name
+
+    def query_str(self, pattern_string):
+        """Run OO client query with a parsed string."""
+        return self.client.query(
+            kcidb.orm.Pattern.parse(pattern_string)
+        )
+
+    def assertNumberOfInstances(self, oo_object_list, num_instances):
```

Comment:
This function doesn't actually care whether "instances", or anything else (like numbers, strings, or types) is contained in the lists. So we can have more generic names for the function and the arguments. This would make the function more useful and the code using it easier to understand.